### PR TITLE
Use runtime ghc libdir for ghc-exactprint and ghc-8.10

### LINKED ISSUE #591: hlint seeking unavailable file /opt/ghc/8.8.4/lib/ghc-8.8.4/settings
### Subject of the issue
hlint seeking unavailabe file /opt/ghc/8.8.4/lib/ghc-8.8.4/settings
and refactor fail since this error

### Your environment
Fedora 33, VSCode 1.51
* Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`
  * 0.6
* Which lsp-client do you use
  * vscode
* Describe your project (alternative: link to the project)
  * single hs, no cradle, no cabal
* Contents of `hie.yaml`
 * N/A

### Steps to reproduce
1. ghcup install ghc under ~/.ghcup
2. open single hs file
3. apply any hint from hlint
4.  prompt of the progress non stop run
5. debug message said /opt/ghc/8.8.4/lib/ghc-8.8.4/settings file not available,
5. symbolic link ~/.ghcup/ghc to /opt/ghc problem fix

### Expected behaviour
seeking a correct ghc installed

### Actual behaviour
seeking  /opt/ghc/8.8.4/lib/ghc-8.8.4/settings

### Include debug information
N/A

### PR DESCRIPTION
**Pull request by jneira**
* We were using it only in the code path for ghc < 8.10
* But I guess some change in either `ghc-exacprint` or `apply-refact` has exposed the ghc-wrapper  for the code path used in ghc-8.10 here (`applyRefact'`)
* Fixes #591

### CODE REVIEW

**Comment by jneira** [comment]
This *must* have a regression test but it would suppose:
* using ghc inits default location to install hls
* moving the ghc installation to other dir
* set that location in path
* run the tests that might be broken for the previous step
* restore the original ghc installation path

Other option could be test the executables, being careful to change the ghc installation, see #595

**Comment by jneira** [comment]
I have to update stack.yaml's

**Review by Ailrun — COMMENTED** [review]
Thanks, waiting for regression tests and stack.yaml updates...

File: plugins/hls-hlint-plugin/hls-hlint-plugin.cabal [hunk 1]
```diff
@@ -53,13 +53,13 @@ library
...
 
-  if ((!flag(ghc-lib) && impl(ghc >=8.10.1)) && impl(ghc <8.11.0))
+  if (!flag(ghc-lib) && impl(ghc >=8.10.1) && impl(ghc <9.0.0))
     build-depends: ghc ^>= 8.10
 
   else
     build-depends:
       , ghc
-      , ghc-lib            ^>= 8.10.2.20200916
+      , ghc-lib            ^>= 8.10.4.20210206
```
Comment (Ailrun) [inline plugins/hls-hlint-plugin/hls-hlint-plugin.cabal:62]: Is it OK to not update cabal.project index?

Comment (jneira) [reply plugins/hls-hlint-plugin/hls-hlint-plugin.cabal:62]: Yeah, that ghc-lib version was releases on 2021-02-06T16:51:55Z

**Comment by jneira** [comment]
> 
> 
> Thanks, waiting for regression tests and stack.yaml updates...

If you dont mind i would prefer to merge this without the test, to fix the issue asap.
That test is more complex than the regular ones and maybe the right way would be #595

**Review by Ailrun — APPROVED** [review]

**Pull request merged**